jax/experimental/pjit.py

```diff
@@ -322,6 +325,31 @@ def _pjit_init_to_final_params(params):
   return bind_params
 core.initial_to_final_param_rules[pjit_call_p] = _pjit_init_to_final_params
 
+def _check_resources_against_named_axes(what, aval, pos_axis_resources, named_axis_resources):
+  pjit_resources = set(it.chain.from_iterable(pos_axis_resources))
+  aval_resources = set(it.chain.from_iterable(
+    named_axis_resources[a] for a in aval.named_shape))
+  overlap = pjit_resources & aval_resources
+  if overlap:
+    raise JAXTypeError(
+        f"{what} has an axis resources specification of {pos_axis_resources} "
+        f"which overlaps with resources partitioning named dimensions of "
+        f"that value (both use mesh axes {maps.show_axes(overlap)})")
```

Comment:
Maybe it's just me, but it took me a long time to parse "resources partitioning named dimensions" correctly :P 
I think "resources that partition named dimensions" would have helped. I'm also still not sure what "that value" is.

I also feel like xmap should be mentioned somewhere in here, maybe:
"{what} has an axis resources specification of {...} that uses one or more mesh axes already used by xmap to partition a named axis (both use mesh axes {...})"
You could even leave out the "to partition a named axis" bit and just leave it at "already used by xmap".